netboot/main.go

```diff
@@ -165,15 +171,25 @@ func boot(ifname string, dhcp dhcpFunc) error {
 	}
 	debug("DHCP: boot file URL is %s", bootfile)
 	// check for supported schemes
-	if !strings.HasPrefix(bootfile, "http://") {
-		return fmt.Errorf("DHCP: can only handle http scheme")
+	scheme, err := getScheme(bootfile)
+	if scheme == "" || err != nil {
+		return errors.New("DHCP: can only handle http/https scheme")
 	}
 
+	client, err := getClientForBootfile(bootfile)
+	if err != nil {
+		return fmt.Errorf("DHCP: cannot get http client for %s: %v", bootfile, err)
+	}
 	log.Printf("DHCP: fetching boot file URL: %s", bootfile)
+
 	var resp *http.Response
 	for attempt := 0; attempt < maxHTTPAttempts; attempt++ {
 		log.Printf("netboot: attempt %d for http.Get", attempt+1)
-		resp, err = http.Get(bootfile)
+		req, err := http.NewRequest(http.MethodGet, bootfile, nil)
```

Comment:
you reuse `http.MethodGet` twice, this would be better off in its own variable `method := http.MethodGet` used here and below

Reply:
I'll actually make the log message less verbose. We shouldn't care that it's a GET method.